extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/MicrometerConfigUnremovableProcessor.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.micrometer.deployment;
+
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+
+public class MicrometerConfigUnremovableProcessor {
+
+    /**
+     * config objects are beans, but they are not unremovable by default
+     */
+    @BuildStep
+    UnremovableBeanBuildItem mpConfigAsBean() {
+        return UnremovableBeanBuildItem.beanTypes(MicrometerConfig.class);
+    }
+
```

Comment:
If we rename this class... "AlwaysOnProcessor?!?!" .. the feature thing could go here.

Reply:
Aside, I have removed the 🤬 service loader here (for other reasons): https://github.com/quarkusio/quarkus/pull/26978.